Automattic/vip-workflow-plugin
# Refactor the remaining modules

## Description

This is a big PR and needs to be reviewed and tested thoroughly. I have refactored the remaining modules to be independent, so all the remaining Edit Flow logic is gone. I've left the folder structure in place as that felt like it'd be way too much. That would be the next item on the list imo.

I have also updated the minimum PHP version to 8.1 and the minimum WordPress version to 6.4.

Module specific changes:

Custom Status: I've made the pending review status mandatory. That felt like it's necessary because some weird bugs were beginning to show up where the pending status was being duplicated. If no statuses are present, we assume just draft and pending are allowed and those are the default statuses in core to begin with. So I've left that as is. I've also shifted all the core hacks away to their own class.

Settings: Everything is isolated to this module, but all options related operations happen through one utility. It ensures that the default option values exist and if we do enable migrations that can go there. I've also updated the settings updated notice to be an admin banner similar to our regular notice.

Items remaining that we need to tackle for the future:

Supporting post types beyond the two default ones is not present right now. Technically we didn't have it working properly before, because we deleted the Edit Flow code. So we need to that back in to ensure we support that.

Fixes #45, #46, and #49 

## Steps to Test

- Setup the plugin from scratch (delete everything)
- Ensure all the defaults get setup
- Ensure everything works
